src/Foundation/Console/InfoCommand.php

```diff
@@ -126,4 +137,28 @@ private function findPackageVersion($path, $fallback = null)
 
         return $fallback;
     }
+
+    private function identifyQueueDriver(): string
+    {
+        // Get instantiated queue class
+        $queue = resolve(Queue::class);
+        // Get class name
+        $queue = get_class($queue);
+        // Drop the namespace
+        $queue = Str::afterLast($queue, '\\');
+        // Lowercase the class name
+        $queue = strtolower($queue);
+        // Drop everything like queue SyncQueue, RedisQueue
+        $queue = str_replace('queue', null, $queue);
+
+        return $queue;
+    }
+
+    private function identifyDatabaseVersion(): string
+    {
+        /** @var MySqlConnection $connection */
+        $connection = resolve(ConnectionInterface::class);
```

Comment:
why aren't we injecting this instead just like the settings repository ?

Reply:
Yeah I wasn't sure about that either. I think either we inject all of them or none. The drawback of injecting them in the construct is that they will be resolved when the command is registered, not just when executed. So best would be to delay resolving as long as possible. I guess after Settings I switched that strategy.

Reply:
> The drawback of injecting them in the construct is that they will be resolved when the command is registered

The commands are only resolved after the entire application has booted, right before symfony console application takes over, so I don't believe there will be any issues with injecting them.
https://github.com/flarum/core/blob/4acff91f8063fcced9bf8c9a76fbb510d06823c0/src/Console/Server.php#L30-L38

Reply:
Thanks 😍 great feedback!